test/unit-test/FreeRTOS_IPv6_Utils/FreeRTOS_IPv6_Utils_utest.c

```diff
@@ -0,0 +1,367 @@
+
+    usReturn = prvChecksumIPv6Checks( pucEthernetBuffer, uxBufferLength, &xSet );
+
+    /* Return value other than zero implies length check fail */
+    TEST_ASSERT_EQUAL( 1, usReturn );
+    TEST_ASSERT_EQUAL( ipINVALID_LENGTH, xSet.usChecksum );
+}
+
+/**
+ * @brief This function validate handling case of
+ *        incomplete IPv6 packet.
+ */
+void test_prvChecksumIPv6Checks_IncompleteIPv6Packet( void )
+{
+    uint16_t usReturn;
+    uint8_t pucEthernetBuffer[ ipconfigTCPv6_MSS ];
+    IPHeader_IPv6_t * pxIPv6Packet;
+    size_t uxBufferLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER;
+    struct xPacketSummary xSet;
+
+    memset( pucEthernetBuffer, 0, ipconfigTCPv6_MSS );
+
+    pxIPv6Packet = ( IPHeader_IPv6_t * ) pucEthernetBuffer;
+    pxIPv6Packet->usPayloadLength = ipSIZE_OF_IPv6_PAYLOAD_LEN;
+    xSet.pxIPPacket_IPv6 = ( ( const IPHeader_IPv6_t * ) pucEthernetBuffer );
+
+    usReturn = prvChecksumIPv6Checks( pucEthernetBuffer, uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( 2, usReturn );
+    TEST_ASSERT_EQUAL( ipINVALID_LENGTH, xSet.usChecksum );
+}
+
+/**
+ * @brief This function validate all length checks passing
+ *        successfully.
+ */
+void test_prvChecksumIPv6Checks_Success( void )
+{
+    uint16_t usReturn;
+    uint8_t pucEthernetBuffer[ ipconfigTCPv6_MSS ];
+    IPHeader_IPv6_t * pxIPv6Packet;
+    size_t uxBufferLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + 1;
+    struct xPacketSummary xSet;
+
+    memset( pucEthernetBuffer, 0, ipconfigTCPv6_MSS );
+
+    pxIPv6Packet = ( IPHeader_IPv6_t * ) pucEthernetBuffer;
+    xSet.pxIPPacket_IPv6 = ( ( const IPHeader_IPv6_t * ) pucEthernetBuffer );
+
+    usReturn = prvChecksumIPv6Checks( pucEthernetBuffer, uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( pdFALSE, usReturn );
+}
+
+/*
+ * ===================================================
+ *        Test for prvChecksumICMPv6Checks
+ * ===================================================
+ */
+
+
+/**
+ * @brief This function verify sending an invalid length.
+ */
+void test_prvChecksumICMPv6Checks_Default_InvalidLength( void )
+{
+    uint16_t usReturn;
+    size_t uxBufferLength = ipSIZE_OF_ETH_HEADER + ipSIZE_OF_IPv6_HEADER + ipSIZE_OF_ICMPv6_HEADER - 1;
+    struct xPacketSummary xSet;
+    ICMPHeader_IPv6_t xICMPHeaderIPv6;
+    ProtocolHeaders_t xProtocolHeaders;
+
+    memset( &xSet, 0, sizeof( struct xPacketSummary ) );
+    xICMPHeaderIPv6.ucTypeOfMessage = ipICMP_Incorrect_Msg_IPv6;
+    xProtocolHeaders.xICMPHeaderIPv6 = xICMPHeaderIPv6;
+    xSet.uxIPHeaderLength = ipSIZE_OF_IPv6_HEADER;
+    xSet.pxProtocolHeaders = &xProtocolHeaders;
+
+    usReturn = prvChecksumICMPv6Checks( uxBufferLength, &xSet );
+
+    TEST_ASSERT_EQUAL( ipFail_INVALID_LENGTH, usReturn );
+    TEST_ASSERT_EQUAL( ipINVALID_LENGTH, xSet.usChecksum );
+}
+
+/**
+ * @brief This function verify sending an valid length.
+ */
+void test_prvChecksumICMPv6Checks_Default_ValidLength( void )
+{
+    uint16_t usReturn;
```

Comment:
The return type of function "prvChecksumICMPv6Checks" is BaseType_t . Why are we defining the return type as "uint16_t". This is a standard case where if BaseType_t  definition is changed, the tests will start failing.

Reply:
Updated